internal/subshell/pwsh/pwsh.go

```diff
@@ -0,0 +1,169 @@
+package pwsh
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/osutils"
+	"github.com/ActiveState/cli/internal/osutils/user"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/subshell/cmd"
+	"github.com/ActiveState/cli/internal/subshell/sscommon"
+	"github.com/ActiveState/cli/pkg/project"
+)
+
+var escaper *osutils.ShellEscape
+
+func init() {
+	escaper = osutils.NewBatchEscaper()
+}
+
+// SubShell covers the subshell.SubShell interface, reference that for documentation
+type SubShell struct {
+	binary string
+	rcFile *os.File
+	cmd    *exec.Cmd
+	env    map[string]string
+	errs   chan error
+}
+
+const Name string = "powershell"
+
+// Shell - see subshell.SubShell
+func (v *SubShell) Shell() string {
+	return Name
+}
+
+// Binary - see subshell.SubShell
+func (v *SubShell) Binary() string {
+	return v.binary
+}
+
+// SetBinary - see subshell.SubShell
+func (v *SubShell) SetBinary(binary string) {
+	v.binary = binary
+}
+
+// WriteUserEnv - see subshell.SubShell
+func (v *SubShell) WriteUserEnv(cfg sscommon.Configurable, env map[string]string, envType sscommon.RcIdentification, userScope bool) error {
+	cmdShell := &cmd.SubShell{}
+	if err := cmdShell.WriteUserEnv(cfg, env, envType, userScope); err != nil {
+		return errs.Wrap(err, "Forwarded WriteUserEnv call failed")
+	}
```

Comment:
Super dirty, but all of this subshell stuff desperately needs a refactor, I didn't see much value in trying to get something satisfying to fit into something .. not (impossible).

Reply:
Do we have a ticket for refactoring all of this? If not, we may want to file one.